commit a02e2677ab02bcded6111d74320d4c1f4f062798
Author: philippe <philippe@a5019735-40e9-0310-863c-91ae7b9d1cf9>
Date:   Wed Mar 6 22:39:18 2013 +0000

    Fix 316144 (valgrind.1 manpage contains ??? strings for references)
    Patch by Mark Wielaard.
    
    
    git-svn-id: svn://svn.valgrind.org/valgrind/trunk@13314 a5019735-40e9-0310-863c-91ae7b9d1cf9

diff --git a/docs/xml/manual-core-adv.xml b/docs/xml/manual-core-adv.xml
index 2a022a0..00376a1 100644
--- a/docs/xml/manual-core-adv.xml
+++ b/docs/xml/manual-core-adv.xml
@@ -293,9 +293,9 @@ tool-specific macros).</para>
 
 
 
-
-<sect1 id="manual-core-adv.gdbserver" 
-       xreflabel="Debugging your program using Valgrind's gdbserver and GDB">
+<!-- Referenced from both the manual and manpage -->
+<sect1 id="&vg-gdbserver-id;"
+       xreflabel="&vg-gdbserver-label;">
 <title>Debugging your program using Valgrind gdbserver and GDB</title>
 
 <para>A program running under Valgrind is not executed directly by the
@@ -381,7 +381,7 @@ and gives the results back to GDB.
 <para>GDB can use various kinds of channels (TCP/IP, serial line, etc)
 to communicate with the gdbserver.  In the case of Valgrind's
 gdbserver, communication is done via a pipe and a small helper program
-called <xref linkend="manual-core-adv.vgdb"/>, which acts as an
+called <xref linkend="&vg-vgdb-id;"/>, which acts as an
 intermediary.  If no GDB is in use, vgdb can also be
 used to send monitor commands to the Valgrind gdbserver from a shell
 command line.
@@ -1105,8 +1105,9 @@ $5 = 36
 
 </sect2>
 
-<sect2 id="manual-core-adv.vgdb"
-       xreflabel="vgdb">
+<!-- Referenced from both the manual and manpage -->
+<sect2 id="&vg-vgdb-id;"
+       xreflabel="&vg-vgdb-label;">
 <title>vgdb command line options</title>
 <para> Usage: <computeroutput>vgdb [OPTION]... [[-c] COMMAND]...</computeroutput></para>
 
@@ -1254,8 +1255,9 @@ vgdb v.set log_output -c leak_check any
 </sect2>
 
 
-<sect2 id="manual-core-adv.valgrind-monitor-commands" 
-       xreflabel="Valgrind monitor commands">
+<!-- Referenced from both the manual and manpage -->
+<sect2 id="&vg-monitor-id;"
+       xreflabel="&vg-monitor-label;">
 <title>Valgrind monitor commands</title>
 
 <para>This section describes the Valgrind monitor commands, available
diff --git a/docs/xml/manual-core.xml b/docs/xml/manual-core.xml
index 70b0989..a60c3a4 100644
--- a/docs/xml/manual-core.xml
+++ b/docs/xml/manual-core.xml
@@ -152,7 +152,8 @@ likely.</para>
 </sect1>
 
 
-<sect1 id="manual-core.comment" xreflabel="The Commentary">
+<!-- Referenced from both the manual and manpage -->
+<sect1 id="&vg-comment-id;" xreflabel="&vg-comment-label;">
 <title>The Commentary</title>
 
 <para>Valgrind tools write a commentary, a stream of text, detailing
@@ -857,7 +858,7 @@ in most cases.  We group the available options by rough categories.</para>
       be used in conjunction with the
       <computeroutput>valgrind-listener</computeroutput> program.  For
       further details, see 
-      <link linkend="manual-core.comment">the commentary</link>
+      <link linkend="&vg-comment-id;">the commentary</link>
       in the manual.</para>
     </listitem>
   </varlistentry>
@@ -1755,7 +1756,7 @@ need to use them.</para>
       between fairness and performance. For more details about the
       Valgrind thread serialisation scheme and its impact on
       performance and thread scheduling, see
-      <xref linkend="manual-core.pthreads_perf_sched"/>.
+      <xref linkend="&vg-pthreads-perf-sched-id;"/>.
 
       <itemizedlist>
         <listitem> <para>The value <option>--fair-sched=yes</option>
@@ -2084,7 +2085,8 @@ everything is shared (a thread) or nothing is shared (fork-like); partial
 sharing will fail.
 </para>
 
-<sect2 id="manual-core.pthreads_perf_sched" xreflabel="Scheduling and Multi-Thread Performance">
+<!-- Referenced from both the manual and manpage -->
+<sect2 id="&vg-pthreads-perf-sched-id;" xreflabel="&vg-pthreads-perf-sched-label;">
 <title>Scheduling and Multi-Thread Performance</title>
 
 <para>A thread executes code only when it holds the abovementioned
diff --git a/docs/xml/valgrind-manpage.xml b/docs/xml/valgrind-manpage.xml
index 8f2f630..bd4b924 100644
--- a/docs/xml/valgrind-manpage.xml
+++ b/docs/xml/valgrind-manpage.xml
@@ -240,7 +240,22 @@ callgrind_annotate(1),
 callgrind_control(1),
 ms_print(1),
 <filename>&vg-docs-path;</filename> or
-<filename>&vg-docs-url;</filename>.
+<filename>&vg-docs-url;</filename>,
+<ulink id="&vg-gdbserver-id;"
+       xreflabel="&vg-gdbserver-label;"
+       url="&vg-gdbserver-url;">&vg-gdbserver-label;</ulink>
+<ulink id="&vg-vgdb-id;"
+       xreflabel="&vg-vgdb-label;"
+       url="&vg-vgdb-url;">&vg-vgdb-label;</ulink>,
+<ulink id="&vg-monitor-id;"
+       xreflabel="&vg-monitor-label;"
+       url="&vg-monitor-url;">&vg-monitor-label;</ulink>,
+<ulink id="&vg-comment-id;"
+       xreflabel="&vg-comment-label;"
+       url="&vg-comment-url;">&vg-comment-label;</ulink>,
+<ulink id="&vg-pthreads-perf-sched-id;"
+       xreflabel="&vg-pthreads-perf-sched-label;"
+       url="&vg-pthreads-perf-sched-url;">&vg-pthreads-perf-sched-label;</ulink>.
 </para>
 
 </refsect1>
diff --git a/docs/xml/vg-entities.xml b/docs/xml/vg-entities.xml
index e65ed3e..f5994c1 100644
--- a/docs/xml/vg-entities.xml
+++ b/docs/xml/vg-entities.xml
@@ -23,4 +23,38 @@
 <!ENTITY vg-docs-url   "http://www.valgrind.org/docs/manual/index.html">
 <!ENTITY cl-gui-url    "http://kcachegrind.sourceforge.net/cgi-bin/show.cgi/KcacheGrindIndex">
 
+<!-- Some references are used between core manual manual and manpages. -->
+<!-- Define them here so the can easily be used in both places.  -->
+<!-- See manual-core*.xml and valgrind-manpage.xml -->
+<!ENTITY vg-manual-url "http://www.valgrind.org/docs/manual/">
+
+<!ENTITY vg-gdbserver-manual "manual-core-adv">
+<!ENTITY vg-gdbserver-ref "gdbserver">
+<!ENTITY vg-gdbserver-label "Debugging your program using Valgrind's gdbserver and GDB">
+<!ENTITY vg-gdbserver-id "&vg-gdbserver-manual;.&vg-gdbserver-ref;">
+<!ENTITY vg-gdbserver-url "&vg-manual-url;&vg-gdbserver-manual;.html#&vg-gdbserver-id;">
+
+<!ENTITY vg-vgdb-manual "manual-core-adv">
+<!ENTITY vg-vgdb-ref "vgdb">
+<!ENTITY vg-vgdb-label "vgdb">
+<!ENTITY vg-vgdb-id "&vg-vgdb-manual;.&vg-vgdb-ref;">
+<!ENTITY vg-vgdb-url "&vg-manual-url;&vg-vgdb-manual;.html#&vg-vgdb-id;">
+
+<!ENTITY vg-monitor-manual "manual-core-adv">
+<!ENTITY vg-monitor-ref "valgrind-monitor-commands">
+<!ENTITY vg-monitor-label "Valgrind monitor commands">
+<!ENTITY vg-monitor-id "&vg-monitor-manual;.&vg-monitor-ref;">
+<!ENTITY vg-monitor-url "&vg-manual-url;&vg-monitor-manual;.html#&vg-monitor-id;">
+
+<!ENTITY vg-comment-manual "manual-core">
+<!ENTITY vg-comment-ref "comment">
+<!ENTITY vg-comment-label "The Commentary">
+<!ENTITY vg-comment-id "&vg-comment-manual;.&vg-comment-ref;">
+<!ENTITY vg-comment-url "&vg-manual-url;&vg-comment-manual;.html#&vg-comment-id;">
+
+<!ENTITY vg-pthreads-perf-sched-manual "manual-core">
+<!ENTITY vg-pthreads-perf-sched-ref "pthreads_perf_sched">
+<!ENTITY vg-pthreads-perf-sched-label "Scheduling and Multi-Thread Performance">
+<!ENTITY vg-pthreads-perf-sched-id "&vg-pthreads-perf-sched-manual;.&vg-pthreads-perf-sched-ref;">
+<!ENTITY vg-pthreads-perf-sched-url "&vg-manual-url;&vg-pthreads-perf-sched-manual;.html#&vg-pthreads-perf-sched-id;">
 

commit 90285ba7ed34f3fdef235d2976399bcd360d4515
Author: philippe <philippe@a5019735-40e9-0310-863c-91ae7b9d1cf9>
Date:   Sun Mar 10 16:20:10 2013 +0000

    fix 315959 valgrind man page has bogus SGCHECK (and no BBV) OPTIONS section
    PAtch from Mark Wielaard.
    
    
    
    git-svn-id: svn://svn.valgrind.org/valgrind/trunk@13323 a5019735-40e9-0310-863c-91ae7b9d1cf9

diff --git a/exp-sgcheck/docs/sg-manual.xml b/exp-sgcheck/docs/sg-manual.xml
index 8632199..8b717e8 100644
--- a/exp-sgcheck/docs/sg-manual.xml
+++ b/exp-sgcheck/docs/sg-manual.xml
@@ -31,7 +31,7 @@ observation about the likely forms of stack and global array accesses.
 <sect1 id="sg-manual.options" xreflabel="SGCheck Command-line Options">
 <title>SGCheck Command-line Options</title>
 
-<para>There are no SGCheck-specific command-line options at present.</para>
+<para id="sg.opts.list">There are no SGCheck-specific command-line options at present.</para>
 <!--
 <para>SGCheck-specific command-line options are:</para>
 

commit e6bce130ae5da444da196cac75102554e80e9ce0
Author: philippe <philippe@a5019735-40e9-0310-863c-91ae7b9d1cf9>
Date:   Sun Mar 10 16:29:02 2013 +0000

    Fix 316145 - callgrind command line options in manpage reference (unknown) callgrind manual
    Patch by Mark Wielaard.
    
    
    
    git-svn-id: svn://svn.valgrind.org/valgrind/trunk@13324 a5019735-40e9-0310-863c-91ae7b9d1cf9

diff --git a/cachegrind/docs/cg-manual.xml b/cachegrind/docs/cg-manual.xml
index fb6f978..13331d8 100644
--- a/cachegrind/docs/cg-manual.xml
+++ b/cachegrind/docs/cg-manual.xml
@@ -3,8 +3,8 @@
   "http://www.oasis-open.org/docbook/xml/4.2/docbookx.dtd"
 [ <!ENTITY % vg-entities SYSTEM "../../docs/xml/vg-entities.xml"> %vg-entities; ]>
 
-
-<chapter id="cg-manual" xreflabel="Cachegrind: a cache and branch-prediction profiler">
+<!-- Referenced from both the manual and manpage -->
+<chapter id="&vg-cg-manual-id;" xreflabel="&vg-cg-manual-label;">
 <title>Cachegrind: a cache and branch-prediction profiler</title>
 
 <para>To use this tool, you must specify
diff --git a/callgrind/docs/cl-manual.xml b/callgrind/docs/cl-manual.xml
index ab8d9bb..2f08ac8 100644
--- a/callgrind/docs/cl-manual.xml
+++ b/callgrind/docs/cl-manual.xml
@@ -88,7 +88,7 @@ optimization opportunities depend on changing code in the callers, in
 particular by reducing the call count.</para>
 
 <para>Callgrind's cache simulation is based on that of Cachegrind.
-Read the documentation for <xref linkend="cg-manual"/> first.  The material
+Read the documentation for <xref linkend="&vg-cg-manual-id;"/> first.  The material
 below describes the features supported in addition to Cachegrind's
 features.</para>
 
@@ -936,7 +936,7 @@ Also see <xref linkend="cl-manual.cycles"/>.</para>
       Cache misses on instruction reads ("I1mr"/"ILmr"),
       data read accesses ("Dr") and related cache misses ("D1mr"/"DLmr"),
       data write accesses ("Dw") and related cache misses ("D1mw"/"DLmw").
-      For more information, see <xref linkend="cg-manual"/>.
+      For more information, see <xref linkend="&vg-cg-manual-id;"/>.
       </para>
     </listitem>
   </varlistentry>
diff --git a/docs/xml/valgrind-manpage.xml b/docs/xml/valgrind-manpage.xml
index bd4b924..a4b84ff 100644
--- a/docs/xml/valgrind-manpage.xml
+++ b/docs/xml/valgrind-manpage.xml
@@ -255,7 +255,10 @@ ms_print(1),
        url="&vg-comment-url;">&vg-comment-label;</ulink>,
 <ulink id="&vg-pthreads-perf-sched-id;"
        xreflabel="&vg-pthreads-perf-sched-label;"
-       url="&vg-pthreads-perf-sched-url;">&vg-pthreads-perf-sched-label;</ulink>.
+       url="&vg-pthreads-perf-sched-url;">&vg-pthreads-perf-sched-label;</ulink>,
+<ulink id="&vg-cg-manual-id;"
+       xreflabel="&vg-cg-manual-label;"
+       url="&vg-cg-manual-url;">&vg-cg-manual-label;</ulink>.
 </para>
 
 </refsect1>
diff --git a/docs/xml/vg-entities.xml b/docs/xml/vg-entities.xml
index f5994c1..29a337f 100644
--- a/docs/xml/vg-entities.xml
+++ b/docs/xml/vg-entities.xml
@@ -58,3 +58,7 @@
 <!ENTITY vg-pthreads-perf-sched-id "&vg-pthreads-perf-sched-manual;.&vg-pthreads-perf-sched-ref;">
 <!ENTITY vg-pthreads-perf-sched-url "&vg-manual-url;&vg-pthreads-perf-sched-manual;.html#&vg-pthreads-perf-sched-id;">
 
+
+<!ENTITY vg-cg-manual-id "cg-manual">
+<!ENTITY vg-cg-manual-label "Cachegrind: a cache and branch-prediction profiler">
+<!ENTITY vg-cg-manual-url "&vg-manual-url;&vg-cg-manual-id;.html">
diff -ur valgrind-3.8.1.orig/docs/callgrind_annotate.1 valgrind-3.8.1/docs/callgrind_annotate.1
--- valgrind-3.8.1.orig/docs/callgrind_annotate.1	2013-03-12 14:42:11.627471322 +0100
+++ valgrind-3.8.1/docs/callgrind_annotate.1	2013-03-12 14:42:28.762370486 +0100
@@ -1,13 +1,13 @@
 '\" t
 .\"     Title: Callgrind Annotate
 .\"    Author: [see the "Author" section]
-.\" Generator: DocBook XSL Stylesheets v1.75.2 <http://docbook.sf.net/>
-.\"      Date: 09/18/2012
+.\" Generator: DocBook XSL Stylesheets v1.77.1 <http://docbook.sf.net/>
+.\"      Date: 03/12/2013
 .\"    Manual: Release 3.8.0
 .\"    Source: Release 3.8.0
 .\"  Language: English
 .\"
-.TH "CALLGRIND ANNOTATE" "1" "09/18/2012" "Release 3.8.0" "Release 3.8.0"
+.TH "CALLGRIND ANNOTATE" "1" "03/12/2013" "Release 3.8.1" "Release 3.8.1"
 .\" -----------------------------------------------------------------
 .\" * Define some portability stuff
 .\" -----------------------------------------------------------------
diff -ur valgrind-3.8.1.orig/docs/callgrind_control.1 valgrind-3.8.1/docs/callgrind_control.1
--- valgrind-3.8.1.orig/docs/callgrind_control.1	2013-03-12 14:42:11.630471305 +0100
+++ valgrind-3.8.1/docs/callgrind_control.1	2013-03-12 14:42:28.778370392 +0100
@@ -1,13 +1,13 @@
 '\" t
 .\"     Title: Callgrind Control
 .\"    Author: [see the "Author" section]
-.\" Generator: DocBook XSL Stylesheets v1.75.2 <http://docbook.sf.net/>
-.\"      Date: 09/18/2012
+.\" Generator: DocBook XSL Stylesheets v1.77.1 <http://docbook.sf.net/>
+.\"      Date: 03/12/2013
 .\"    Manual: Release 3.8.0
 .\"    Source: Release 3.8.0
 .\"  Language: English
 .\"
-.TH "CALLGRIND CONTROL" "1" "09/18/2012" "Release 3.8.0" "Release 3.8.0"
+.TH "CALLGRIND CONTROL" "1" "03/12/2013" "Release 3.8.1" "Release 3.8.1"
 .\" -----------------------------------------------------------------
 .\" * Define some portability stuff
 .\" -----------------------------------------------------------------
diff -ur valgrind-3.8.1.orig/docs/cg_annotate.1 valgrind-3.8.1/docs/cg_annotate.1
--- valgrind-3.8.1.orig/docs/cg_annotate.1	2013-03-12 14:42:11.627471322 +0100
+++ valgrind-3.8.1/docs/cg_annotate.1	2013-03-12 14:42:28.745370586 +0100
@@ -1,13 +1,13 @@
 '\" t
 .\"     Title: cg_annotate
 .\"    Author: [see the "Author" section]
-.\" Generator: DocBook XSL Stylesheets v1.75.2 <http://docbook.sf.net/>
-.\"      Date: 09/18/2012
+.\" Generator: DocBook XSL Stylesheets v1.77.1 <http://docbook.sf.net/>
+.\"      Date: 03/12/2013
 .\"    Manual: Release 3.8.0
 .\"    Source: Release 3.8.0
 .\"  Language: English
 .\"
-.TH "CG_ANNOTATE" "1" "09/18/2012" "Release 3.8.0" "Release 3.8.0"
+.TH "CG_ANNOTATE" "1" "03/12/2013" "Release 3.8.1" "Release 3.8.1"
 .\" -----------------------------------------------------------------
 .\" * Define some portability stuff
 .\" -----------------------------------------------------------------
diff -ur valgrind-3.8.1.orig/docs/ms_print.1 valgrind-3.8.1/docs/ms_print.1
--- valgrind-3.8.1.orig/docs/ms_print.1	2013-03-12 14:42:11.627471322 +0100
+++ valgrind-3.8.1/docs/ms_print.1	2013-03-12 14:42:28.793370304 +0100
@@ -1,13 +1,13 @@
 '\" t
 .\"     Title: ms_print
 .\"    Author: [see the "Author" section]
-.\" Generator: DocBook XSL Stylesheets v1.75.2 <http://docbook.sf.net/>
-.\"      Date: 09/18/2012
+.\" Generator: DocBook XSL Stylesheets v1.77.1 <http://docbook.sf.net/>
+.\"      Date: 03/12/2013
 .\"    Manual: Release 3.8.0
 .\"    Source: Release 3.8.0
 .\"  Language: English
 .\"
-.TH "MS_PRINT" "1" "09/18/2012" "Release 3.8.0" "Release 3.8.0"
+.TH "MS_PRINT" "1" "03/12/2013" "Release 3.8.1" "Release 3.8.1"
 .\" -----------------------------------------------------------------
 .\" * Define some portability stuff
 .\" -----------------------------------------------------------------
diff -ur valgrind-3.8.1.orig/docs/valgrind.1 valgrind-3.8.1/docs/valgrind.1
--- valgrind-3.8.1.orig/docs/valgrind.1	2013-03-12 14:42:11.626471328 +0100
+++ valgrind-3.8.1/docs/valgrind.1	2013-03-12 14:42:28.728370686 +0100
@@ -1,13 +1,13 @@
 '\" t
 .\"     Title: VALGRIND
 .\"    Author: [see the "Author" section]
-.\" Generator: DocBook XSL Stylesheets v1.75.2 <http://docbook.sf.net/>
-.\"      Date: 09/18/2012
+.\" Generator: DocBook XSL Stylesheets v1.77.1 <http://docbook.sf.net/>
+.\"      Date: 03/12/2013
 .\"    Manual: Release 3.8.0
 .\"    Source: Release 3.8.0
 .\"  Language: English
 .\"
-.TH "VALGRIND" "1" "09/18/2012" "Release 3.8.0" "Release 3.8.0"
+.TH "VALGRIND" "1" "03/12/2013" "Release 3.8.1" "Release 3.8.1"
 .\" -----------------------------------------------------------------
 .\" * Define some portability stuff
 .\" -----------------------------------------------------------------
@@ -134,13 +134,13 @@
 is specified\&. This allows an external GNU GDB debugger to control and debug your program when it runs on Valgrind\&.
 \fB\-\-vgdb=full\fR
 incurs significant performance overheads, but provides more precise breakpoints and watchpoints\&. See
-???
+Debugging your program using Valgrind's gdbserver and GDB
 for a detailed description\&.
 .sp
 If the embedded gdbserver is enabled but no gdb is currently being used, the
-???
+vgdb
 command line utility can send "monitor commands" to Valgrind from a shell\&. The Valgrind core provides a set of
-???\&. A tool can optionally provide tool specific monitor commands, which are documented in the tool specific chapter\&.
+Valgrind monitor commands\&. A tool can optionally provide tool specific monitor commands, which are documented in the tool specific chapter\&.
 .RE
 .PP
 \fB\-\-vgdb\-error=<number> [default: 999999999] \fR
@@ -300,6 +300,15 @@
 Specifies an alternative exit code to return if Valgrind reported any errors in the run\&. When set to the default value (zero), the return value from Valgrind will always be the return value of the process being simulated\&. When set to a nonzero value, that value is returned instead, if Valgrind detects any errors\&. This is useful for using Valgrind as part of an automated test suite, since it makes it easy to detect test cases for which Valgrind has reported errors, just by inspecting return codes\&.
 .RE
 .PP
+\fB\-\-sigill\-diagnostics=<yes|no> [default: yes] \fR
+.RS 4
+Enable/disable printing of illegal instruction diagnostics\&. Enabled by default, but defaults to disabled when
+\fB\-\-quiet\fR
+is given\&. The default can always be explicitly overridden by giving this option\&.
+.sp
+When enabled a warning message will be printed with some diagnostics whenever some instruction is encountered that valgrind cannot decode or translate before the program is given a SIGILL signal\&. Often an illegal instruction indicates a bug in the program or missing support for the particular instruction in Valgrind\&. But some programs do deliberately try to execute an instruction that might be missing and trap the SIGILL signal to detect processor features\&.
+.RE
+.PP
 \fB\-\-show\-below\-main=<yes|no> [default: no] \fR
 .RS 4
 By default, stack traces for errors do not show any functions that appear beneath
@@ -755,7 +764,7 @@
 The
 \fB\-\-fair\-sched\fR
 option controls the locking mechanism used by Valgrind to serialise thread execution\&. The locking mechanism controls the way the threads are scheduled, and different settings give different trade\-offs between fairness and performance\&. For more details about the Valgrind thread serialisation scheme and its impact on performance and thread scheduling, see
-???\&.
+Scheduling and Multi-Thread Performance\&.
 .sp
 .RS 4
 .ie n \{\
@@ -1242,7 +1251,7 @@
 \fB\-\-cache\-sim=<yes|no> [default: no] \fR
 .RS 4
 Specify if you want to do full cache simulation\&. By default, only instruction read accesses will be counted ("Ir")\&. With cache simulation, further event counters are enabled: Cache misses on instruction reads ("I1mr"/"ILmr"), data read accesses ("Dr") and related cache misses ("D1mr"/"DLmr"), data write accesses ("Dw") and related cache misses ("D1mw"/"DLmw")\&. For more information, see
-???\&.
+Cachegrind: a cache and branch-prediction profiler\&.
 .RE
 .PP
 \fB\-\-branch\-sim=<yes|no> [default: no] \fR
@@ -1568,7 +1577,9 @@
 \fB--log-file\fR\&.
 .RE
 .SH "SGCHECK OPTIONS"
-<xi:include></xi:include>.SH "BBV OPTIONS"
+.PP
+There are no SGCheck\-specific command\-line options at present\&.
+.SH "BBV OPTIONS"
 .PP
 \fB\-\-bb\-out\-file=<name> [default: bb\&.out\&.%p] \fR
 .RS 4
@@ -1705,9 +1716,46 @@
 cg_annotate(1), callgrind_annotate(1), callgrind_control(1), ms_print(1),
 $INSTALL/share/doc/valgrind/html/index\&.html
 or
-http://www\&.valgrind\&.org/docs/manual/index\&.html\&.
+http://www\&.valgrind\&.org/docs/manual/index\&.html,
+\m[blue]\fBDebugging your program using Valgrind\*(Aqs gdbserver and GDB\fR\m[]\&\s-2\u[1]\d\s+2
+\m[blue]\fBvgdb\fR\m[]\&\s-2\u[2]\d\s+2,
+\m[blue]\fBValgrind monitor commands\fR\m[]\&\s-2\u[3]\d\s+2,
+\m[blue]\fBThe Commentary\fR\m[]\&\s-2\u[4]\d\s+2,
+\m[blue]\fBScheduling and Multi\-Thread Performance\fR\m[]\&\s-2\u[5]\d\s+2,
+\m[blue]\fBCachegrind: a cache and branch\-prediction profiler\fR\m[]\&\s-2\u[6]\d\s+2\&.
 .SH "AUTHOR"
 .PP
 The Valgrind developers\&.
 .PP
 This manpage was written by Andres Roldan <aroldan@debian\&.org> and the Valgrind developers\&.
+.SH "NOTES"
+.IP " 1." 4
+Debugging your program using Valgrind's gdbserver and GDB
+.RS 4
+\%http://www.valgrind.org/docs/manual/manual-core-adv.html#manual-core-adv.gdbserver
+.RE
+.IP " 2." 4
+vgdb
+.RS 4
+\%http://www.valgrind.org/docs/manual/manual-core-adv.html#manual-core-adv.vgdb
+.RE
+.IP " 3." 4
+Valgrind monitor commands
+.RS 4
+\%http://www.valgrind.org/docs/manual/manual-core-adv.html#manual-core-adv.valgrind-monitor-commands
+.RE
+.IP " 4." 4
+The Commentary
+.RS 4
+\%http://www.valgrind.org/docs/manual/manual-core.html#manual-core.comment
+.RE
+.IP " 5." 4
+Scheduling and Multi-Thread Performance
+.RS 4
+\%http://www.valgrind.org/docs/manual/manual-core.html#manual-core.pthreads_perf_sched
+.RE
+.IP " 6." 4
+Cachegrind: a cache and branch-prediction profiler
+.RS 4
+\%http://www.valgrind.org/docs/manual/cg-manual.html
+.RE
